Include/criware/criware_server.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+* Copyright (C) 2022 Gemini
+* ===========================================================
+* File server module
+* -----------------------------------------------------------
+* This module uses a thread to fetch data to DirectSound
+* buffers for ADX and AIX playback.
+* ===========================================================
+*/
+#include "criware.h"
+
+static HANDLE hServer;
+static bool loop = true;
+static CRITICAL_SECTION ADX_crit;
+static int lock_cnt = 0;
+
+//
+void ADX_lock_init()
+{
+	InitializeCriticalSection(&ADX_crit);
+}
+
+void ADX_lock_close()
+{
+	DeleteCriticalSection(&ADX_crit);
+}
+
+void ADX_lock()
+{
+	if(lock_cnt == 0)
+		EnterCriticalSection(&ADX_crit);
+	lock_cnt++;
+}
+
+void ADX_unlock()
+{
+	lock_cnt--;
+	if(lock_cnt == 0)
+		LeaveCriticalSection(&ADX_crit);
+}
+
+DWORD WINAPI server_thread(LPVOID params)
+{
+	while (loop)
+	{
+		adxs_Update();
+		Sleep(1);	// just give the thread enough time for locking from external operations
+					// used to be 10 milliseconds

```

Comment:
Why not use `Sleep(0);` ?  This way it returns to the thread as soon as other threads are done.

See:  https://stackoverflow.com/questions/3727420/significance-of-sleep0